server/build-menu.js

```diff
@@ -0,0 +1,98 @@
+
+const electron = require('electron');
+// Module to control application life.
+const { app, ipcMain } = electron;
+// Module to create native browser window.
+const BrowserWindow = electron.BrowserWindow;
+const Menu = electron.Menu;
+const MenuItem = electron.MenuItem;
+
+const defaultMenuTemplate = [
+  {
+    label: 'File',
+    submenu: [
+      { label: 'Preferences' },
+      { type: 'separator' },
+      { label: 'Close Conversation', accelerator: 'CmdOrCtrl+W' },
+      { label: 'Close Window', accelerator: 'CmdOrCtrl+Shift+W' },
+      { type: 'separator' },
+      { label: 'Quit', accelerator: 'CmdOrCtrl+Q', role: 'quit' },
+    ]
+  },
+  {
+    label: 'Edit',
+    submenu: [
+      { role: 'undo' },
+      { role: 'redo' },
+      { type: 'separator' },
+      { role: 'cut' },
+      { role: 'copy' },
+      { role: 'paste' },
+      { role: 'delete' },
+      { role: 'selectall' },
+      { type: 'separator' },
+      { label: 'Find...', accelerator: 'CmdOrCtrl+F' },
+    ],
+  },
+  {
+    label: 'Help',
+    submenu: [
+      { 'label': 'Keyboard Shortcuts' },
+      { type: 'separator' },
+      { 'label': 'Report Issue...' },
+      { type: 'separator' },
+      { 'label': 'About Operator...' },
+    ],
+  }
+];
+
+function buildMenu (template = defaultMenuTemplate) {
+  let handleMenuClick = (menu) => {
+    if ('submenu' in menu) {
```

Comment:
wow I've never used `in` like this before. Always used `Object.hasOwnProperty()`. This approach can be a little risky since it goes over inherited properties too but maybe that's ok in this case?

Reply:
not really sure; this is just @mihok's code moved into another file. Let's leave it for now.

Reply:
Honestly it was over a year ago and I cant remember if I wrote this by hand or copied it out of a tutorial :woman_shrugging:

Reply:
It looks like pretty horrible code, I may have wrote it myself...

Reply:
who cares it works for now